src/parser.rs

```diff
@@ -165,7 +164,7 @@ impl<'a> Parser<'a> {
     }
 
     /// `function_call` method handles function calls.
-    fn function_call(&mut self) -> Statement {
+    fn function_call(&mut self) -> Expression {
```

Comment:
I think modify the `function_call` method to accept an expression within the parentheses would be solve the current problems.

```rust
fn function_call(&mut self) -> Expression {
    let function_name = self.expect_identifier();
    self.expect_token(TokenType::OpenParen);
    let argument = self.expression(); // parse the argument as an expression
    self.expect_token(TokenType::CloseParen);
    Expression::FunctionCall(function_name, Box::new(argument))
}
```

It seems like the problem is occurring because the parser is expecting an `identifier` after the `OpenParen`, but it is getting a `Number` instead.

The function `expect_identifier` might be expecting an identifier, but in this case, I need to parse an expression within the parentheses, not just an identifier.